src/functor.jl

```diff
@@ -48,6 +48,24 @@ function params!(p::Params, x, seen = IdSet())
   end
 end
 
+"""
+    params(model)
+
+Given a model or specific layers from a model, create a `Params` object pointing to its trainable parameters.
+
+This can be used with [`gradient`](@ref), or as input to the [`Flux.train!`](@ref Flux.train!) function.
+
+The behaviour of `params` on custom types can be customized using [`Functor.@functor`](@ref) or [`Flux.trainable`](@ref).
```

Comment:
```suggestion
The behaviour of `params` on custom types can be customized using [`Functors.@functor`](@ref) or [`Flux.trainable`](@ref).
```
Unfortunately `Functors.@functor` won't resolve because it's a cross-package link. Not sure what we can do about that, perhaps a manual link to the latest version of the Functors docs?

Reply:
According to CI there are many such failures, although they do not cause it to fail overall here.

Reply:
Yes, I saw a couple searching for other examples of this. Short of a solution in Documenter.jl, I guess our best option is a relative link?

Reply:
Since the page this wants to link to is here https://fluxml.ai/Flux.jl/stable/models/functors/#Functors.jl I think it ought to be possible at least to have an automatic link to the heading of the page. Maybe that's what you mean? Like https://raw.githubusercontent.com/JuliaLang/julia/master/doc/src/manual/arrays.md it seems you can write e.g.
`## [Functors.jl](@id man-h2-functors)` and then refer to that.

Reply:
https://juliadocs.github.io/Documenter.jl/stable/man/guide/#Cross-Referencing makes it seem like we can do `[Functors.jl](@ref)` in the doc string and it will just work

Reply:
Worth a try. Also not in scope for this PR, but `Functors.@functor` needs to be rendered correctly in the docs: https://fluxml.ai/Functors.jl/stable/api/

Reply:
Also `trainable` isn't documented either. Looks like we'll need a follow-up PR :)

Reply:
Many PR's to come : )